### PR TITLE
Add L1 block hash to query response

### LINKED ISSUE #300: EVM: add L1 block hash to blocks
Right now we have l1 fee rate in evm Block, but it's not hashed along with the Header.

We might want to add l1 fee rate and l1 block hash (or height) directly into the Header. 

Let's decide and execute on this issue

### PR DESCRIPTION
**Pull request by otaliptus**
# Description
eth_getBlockByNumber now returns both l1_fee_rate and l1_block_hash.

Closes #300.

### CODE REVIEW

File: crates/evm/src/evm/primitive_types.rs [hunk 1]
```diff
@@ -82,6 +82,9 @@ pub(crate) struct Block {
     /// L1 fee rate.
     pub(crate) l1_fee_rate: u64,
 
+    /// Last L1 hash.
+    pub(crate) last_l1_hash: B256,
```
Comment (kpp) [inline crates/evm/src/evm/primitive_types.rs:86]:
```suggestion
    /// The hash of L1 block this L2 block is bound to.
    pub(crate) l1_hash: B256,
```

**Review by eyusufatik — APPROVED** [review]

File: crates/evm/src/genesis.rs [hunk 2]
```diff
@@ -174,6 +174,8 @@ impl<C: sov_modules_api::Context> Evm<C> {
         let block = Block {
             header,
             l1_fee_rate: 0,
+            // TODO: Check this for genesis hash - is it completely fine?
+            last_l1_hash: B256::default(),
```
Comment (kpp) [inline crates/evm/src/genesis.rs:178]: l1_hash

File: crates/evm/src/hooks.rs [hunk 3]
```diff
@@ -44,8 +44,8 @@ where
 
         // populate system events
         let mut system_events = vec![];
-        if let Some(last_l1_hash) = self.last_l1_hash.get(working_set) {
-            if last_l1_hash != da_slot_hash {
+        if let Some(l1_hash) = self.last_l1_hash.get(working_set) {
```
Comment (eyusufatik) [inline crates/evm/src/hooks.rs:47]: don't rename here

File: crates/evm/src/query.rs [hunk 4]
```diff
@@ -172,7 +173,16 @@ impl<C: sov_modules_api::Context> Evm<C> {
             uncles: Default::default(),
             transactions,
             withdrawals: Default::default(),
-            other: Default::default(),
+            other: OtherFields::new(BTreeMap::<String, _>::from([
+                (
```
Comment (eyusufatik) [inline crates/evm/src/query.rs:177]: use camelCase here

**Review by eyusufatik — CHANGES_REQUESTED** [review]

**Pull request merged**